### PR TITLE
Cleanup consensus engine tests

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

The consensus engine tests are in a really bad spot. This PR cleans them up a bit. By defining the genesis globally and adding a helper function to generate a chain in a helper function.

These tests are realistically going to be completely re-written... But I want to update these tests incrementally so we can understand what's changing.

## How this was tested

- [X] CI

### CODE REVIEW

File: snow/engine/snowman/transitive_test.go [hunk 1]
```diff
@@ -551,87 +513,41 @@ func TestEngineBlockedIssue(t *testing.T) {
...
-		context.Background(),
-		te.Ctx.NodeID,
-		blk,
-		false,
-		te.metrics.issued.WithLabelValues(unknownSource),
-	))
-	require.NoError(te.QueryFailed(context.Background(), vdr, 1))
-
-	require.Empty(te.blocked)
-}
```
Comment (StephenButtolph) [inline snow/engine/snowman/transitive_test.go:592]: This test made 100% no sense. `te.blocked` is always empty during this test... So really it's just testing that `issue` works... Which we already test in other tests.

File: snow/engine/snowman/transitive_test.go [hunk 2]
```diff
@@ -30,10 +31,36 @@ var (
...
+			ParentV: root.ID(),
+			HeightV: root.Height() + 1,
+			BytesV:  utils.RandomBytes(32),
+		}
+		chain[i] = root
+	}
+	return chain
+}
+
+func setup(t *testing.T, engCfg Config) (ids.NodeID, validators.Manager, *common.SenderTest, *block.TestVM, *Transitive) {
```
Comment (abi87) [inline snow/engine/snowman/transitive_test.go:63]: seems that `setup` is only called by `setupDefaultConfig`. Consider merging the two functions into a single `setup`

File: snow/engine/snowman/transitive_test.go [hunk 3]
```diff
@@ -93,235 +115,212 @@ func setup(t *testing.T, engCfg Config) (ids.NodeID, validators.Manager, *common
...
+		getRequest = &common.Request{
+			NodeID:    nodeID,
+			RequestID: requestID,
+		}
+		require.Equal(peerID, nodeID)
 		require.Contains([]ids.ID{
-			blk.ID(),
-			gBlk.ID(),
+			parent.ID(),
+			GenesisID,
```
Comment (abi87) [inline snow/engine/snowman/transitive_test.go:230]: aren't parent.ID() and GenesisID equal?

File: snow/engine/snowman/transitive_test.go [hunk 4]
```diff
@@ -2540,42 +2146,17 @@ func TestEngineBubbleVotesThroughInvalidBlock(t *testing.T) {
...
-		TestDecidable: choices.TestDecidable{
-			IDV:     ids.GenerateTestID(),
-			StatusV: choices.Processing,
-		},
-		ParentV: blk2.ID(),
-		HeightV: 3,
-		BytesV:  []byte{3},
-	}
+	blks := BuildChain(Genesis, 3)
+	blk1 := blks[0]
```
Comment (abi87) [inline snow/engine/snowman/transitive_test.go:2153]: we're changing the blk1 status here (it was Processing but here it is genesis, which is Accepted. Does this matter?

File: snow/engine/snowman/transitive_test.go (hunk 3 above)
Comment (StephenButtolph) [reply snow/engine/snowman/transitive_test.go:230]: `parent.ID()` and `GenesisID` are not equal.

**Review by abi87 — COMMENTED** [review]
a nit and a couple of questions.
Definitely complex tests, doing a second pass

**Review by abi87 — APPROVED** [review]

File: snow/engine/snowman/transitive_test.go [hunk 5]
```diff
@@ -30,298 +31,289 @@ var (
...
 	}
-	vm.CantShutdown = false
-	require.NoError(transitive.Shutdown(context.Background()))
+
+	require.NoError(engine.Shutdown(context.Background()))
 	require.True(vmShutdownCalled)
 }
 
-func TestEngineAdd(t *testing.T) {
+func TestEngineDropsAttemptToIssueBlockAfterFailedRequest(t *testing.T) {
```
Comment (aaronbuchwald) [inline snow/engine/snowman/transitive_test.go:134]: Why is this change part of this PR?

**Review by JacobEverly — APPROVED** [review]

**Review by aaronbuchwald — APPROVED** [review]

File: snow/engine/snowman/transitive_test.go (hunk 5 above)
Comment (StephenButtolph) [reply snow/engine/snowman/transitive_test.go:134]: This just changed the name of the test, I felt it was more useful of a name then `TestEngineAdd`

File: snow/engine/snowman/transitive_test.go [hunk 6]
```diff
@@ -30,298 +31,289 @@ var (
...
-	return vdr, vals, sender, vm, te, gBlk
-}
-
-func setupDefaultConfig(t *testing.T) (ids.NodeID, validators.Manager, *common.SenderTest, *block.TestVM, *Transitive, snowman.Block) {
-	engCfg := DefaultConfig(t)
-	return setup(t, engCfg)
+	return vdr, config.Validators, sender, vm, te
 }
 
 func TestEngineShutdown(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply snow/engine/snowman/transitive_test.go]: It isn't. Removed.

File: snow/engine/snowman/transitive_test.go [hunk 7]
```diff
@@ -30,298 +31,289 @@ var (
...
 	vm.Default(true)
 	vm.CantSetState = false
 	vm.CantSetPreference = false
 
-	gBlk := &snowman.TestBlock{TestDecidable: choices.TestDecidable{
-		IDV:     Genesis,
-		StatusV: choices.Accepted,
-	}}
-
 	vm.LastAcceptedF = func(context.Context) (ids.ID, error) {
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply snow/engine/snowman/transitive_test.go:96]: Yeah this should be improved. I mainly wanted to unblock the new test in #2956 for this PR, so not trying to do a total re-write here.

File: snow/engine/snowman/transitive_test.go [hunk 8]
```diff
@@ -30,298 +31,289 @@ var (
...
-	asked := new(bool)
-	reqID := new(uint32)
-	sender.SendGetF = func(_ context.Context, inVdr ids.NodeID, requestID uint32, blkID ids.ID) {
-		*reqID = requestID
-		require.False(*asked)
-		*asked = true
-		require.Equal(vdr, inVdr)
-		require.Equal(blk.Parent(), blkID)
+	var request *common.Request
+	sender.SendGetF = func(_ context.Context, nodeID ids.NodeID, requestID uint32, blkID ids.ID) {
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply snow/engine/snowman/transitive_test.go:129]: There are some things that can definitely be factored out... See some of the changes in #2952.

File: snow/engine/snowman/transitive_test.go [hunk 9]
```diff
@@ -30,298 +31,289 @@ var (
...
-		require.False(*queried)
-		*queried = true
-		*queryRequestID = requestID
-		vdrSet := set.Of(vdr)
-		require.Equal(vdrSet, inVdrs)
-		require.Equal(blk.ID(), blockID)
+	// Handling a pull query for [parent] should result in immediately
+	// responding with chits for [Genesis] along with a request for [parent].
+	require.NoError(engine.PullQuery(context.Background(), peerID, 15, parent.ID(), 1))
+	require.True(sendChitsCalled)
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply snow/engine/snowman/transitive_test.go:215]: We could probably re-write the tests to simulate multiple participants. But it feels like that would be better suited to an E2E test

**Pull request merged**